### DISCUSSION TITLE
Super Adventure Island II: SPC files won't load.

### DISCUSSION THREAD

**Discussion by ghost** [vgmtrans/vgmtrans#938] [Music formats]
I'm trying to get the files from Super Adventure Island II to load into VGMTrans but nothing happens. I tried to see if getting the spcs from different websites would work, but it still didn't. I got the first Adventure Island along with II and only the files from the former work.

**Comment by loveemu** [comment]
Super Adventure Island II is not supported at the moment, I suppose. The game's developed by [Make Software](http://gdri.smspower.org/wiki/index.php/Make_Software) and I have never disassembled their sound program yet.

**Comment by KungFuFurby** [comment]
The first Super Adventure Island game falls under the Quintet/Yuzo Koshiro collection. The same build is used by Magic Sword. It has the same VCMD code (within the scope of what I looked up) as the Japanese version of Actraiser, which is also the earliest version used. The US and European version of ActRaiser (and Soul Blader/Soul Blazer, which uses the same build) are a later version.

Now... regarding the sequel... Make Software is an unusual case. I consider this to be N-SPC, but a non-standard implementation of that.

First of all, there is no relation to Super Nichibutsu Mahjong, Super Nichibutsu Mahjong 2 and Super Nichibutsu Mahjong 4. These are just standard N-SPC/Kankichi VCMD code-wise, including the phrase format. I also determined that relevancy to the older Hudson/Make Software? N-SPC variant used in Bill Laimbeer's Combat Basketball and Earth Light is also not exactly relevant.

Instead, I'm referring to these games, roughly in ordering of versioning (indented means build is identical)...
* Takahashi Meijin no Daibouken-jima II/Super Adventure Island II
* Yokoyama Mitsuteru: Sangokushi Bangi: Sugoroku Eiyuuki
* Nomark Baku Haitou: Shijou Saikyou no Jakushi Tatsu
* Super Momotarou Dentetsu DX
  * Momotarou Dentetsu Happy

Super Adventure Island II is the earliest version, and I can confirm the following differences right out of the gate:
* The phrase format is almost completely cut, as only the first entry is read. That means VCMD $00 terminates the track instead.
* VCMDs $EB-$EC are cut (meaning no tremolo), VCMD $F0 is dummied out with a RET opcode (no vibrato fade in), and VCMD $FB is also dummied out with a RET opcode.
* Vibrato (VCMD $E3) I think has a difference in how the rate is handled. Additionally, it doesn't get a big difference if the depth is higher than $F0.
* VCMD $FC is a new command, with everything beyond that not valid.

The other three builds have some additional differences here and there with additional VCMDs and a slight adjustment or two...

**Comment by mikelow** [comment]
@KungFuFurby thanks, as always, for the explaining the heritage and overview of the format.

I'm moving this to the Music Format discussion.
